Trappist/src/Promact.Trappist.Core/Controllers/AccountController.cs

```diff
@@ -8,16 +13,71 @@ namespace Promact.Trappist.Core.Controllers
 {
     public class AccountController : Controller
     {
-        public IActionResult Login()
+        private readonly IAccountRepository _accountRepository;
+        private readonly IStringConstants _stringConstant;
+        public AccountController(IAccountRepository accountRepository, IStringConstants stringConstant)
         {
+            _accountRepository = accountRepository;
+            _stringConstant = stringConstant;
+        }
+        /// <summary>
+        /// this method is used to see the view of login
+        /// </summary>
+        /// <returns>Login form view</returns>
+        [HttpGet]
+        [AllowAnonymous]
+        public IActionResult Login(string returnUrl = null)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Use the overloading mechanism instead of the optional parameters. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://server12:9005/coding_rules#rule_key=csharpsquid%3AS2360)